GrandCentralBoard/Widgets/Bonus/Source/PeopleWithBonususFetchController.swift

```diff
@@ -48,22 +50,26 @@ final class PeopleWithBonusesFetchController {
     }
 
     private func fetchPeopleWithBonuses(startingFromDate date: NSDate = NSDate.init(), fetchedBonuses: [Bonus], completionBlock: (Result<[Person]>) -> Void) {
-        let take = 100
-        let requestTemplate = TimestampableRequestTemplate<BonusesRequestTemplate>.init(requestTemplate: BonusesRequestTemplate(), date: date, take: take)
+        let requestTemplate = TimestampableRequestTemplate(requestTemplate: BonusesRequestTemplate(), date: date, take: self.pageSize)
         
-        requestSender.sendRequestForRequestTemplate(requestTemplate) { [weak self] result in
+        requestSending.sendRequestForRequestTemplate(requestTemplate) { [weak self] result in
             guard let strongSelf = self else {
                 completionBlock(.Failure(PeopleWithBonususFetchControllerError.Cancelled))
                 return
             }
 
             switch result {
             case .Success(let bonuses):
-                var allBonuses: [Bonus] = fetchedBonuses.reverse()
+                var allBonuses: [Bonus] = fetchedBonuses
                 allBonuses.appendContentsOf(bonuses)
 
-                let people = allBonuses.uniqueReceivers(kPreferredPeopleCount)
-                if people.count >= kPreferredPeopleCount || bonuses.count < take {
+                let people = allBonuses.uniqueReceivers(strongSelf.preferredNumberOfPeople)
+
+                for person in people {
+                    print(person.id)
+                }
+
+                if people.count >= self?.preferredNumberOfPeople || bonuses.count < strongSelf.pageSize {
```

Comment:
I'm not in favor of strongSelf entirely. 

If owner of mine is no longer interested in me being alive, he most definitely want me dead and doing some calls (including completion) after my supposed dealloc is spooky and complicates things.